cache-extension-demo/img/Sequence.txt

```diff
@@ -6,7 +6,7 @@ Extension-->Extension: Initialization complete
 Extension->DynamoDB: Initialize cache map and fetch all \nrecords listed in config.yaml if \n CACHE_EXTENSION_INIT_STARTUP=true
 Extension-->Extension: Initialization complete
 Extension-->Extension: Start a HTTP server
-Lambda->Extension: HTTP Get:/parameters/<name> to read from parameters cache
+Lambda->Extension: HTTP Get:/parameters?name=<name> to read from parameters cache
 Extension-->Lambda: Check whether item exists in cache and not exired?\nIf yes return it or \nread it from parameter store, cache it and return it
-Lambda->Extension: HTTP Get:/dynamodb/<table><hashkey><rangekey> to read from dynamodb cache
+Lambda->Extension: HTTP Get:/dynamodb?name=<table><hashkey><rangekey> to read from dynamodb cache
```

Comment:
I see hyphens in the sample code (see comment above). What about the hyphens?

Reply:
The cache name is created based on the values specified in `config.yaml`
```yaml
dynamodb:
  - table: DynamoDbTable
    hashkey: pKey
    hashkeytype: S
    hashkeyvalue: pKey1
    sortkey: sKey
    sortkeytype: S
    sortkeyvalue: sKey1
```
So the final cache key/array key will be based on `<table><hashkey><rangekey>`. So if user sends in the same key the cache will be fetched automatically

Here is the go code for that:

```go
// Generate key to store in map based with a format "tableName+"-"+hashKeyValue+"-"+sortKeyValue"
func GetKey(dynamodbConfig DynamodbConfiguration) string {
   var key = dynamodbConfig.Table + "-" + dynamodbConfig.HashKeyValue
   if dynamodbConfig.SortKey != "" {
      key += "-" + dynamodbConfig.SortKeyValue
   }
   return key
}
```

This code will help to read the value from cache, so if the user is sending the write key it will be an array lookup, like below:

```go
// Fetch Dynamodb cache
func GetDynamodbCache(name string) string {
   var dbCache = dynamoDbCache[name]
   // If expired or not available in cache then read it from Dynamodb, else return from cache
   if dbCache.CacheData.Data == "" || IsExpired(dbCache.CacheData.CacheExpiry) {
      return GetData(dynamoDbCache[name].DynamodbConfiguration)
   } else {
      return dynamoDbCache[name].CacheData.Data
   }
}
```